fuel-merkle/src/binary/merkle_tree.rs

```diff
@@ -370,7 +390,7 @@ where
 {
     let mut head = subtree.clone();
     while let Some(mut head_next) = head.take_next() {
-        head = join_subtrees(&mut head_next, &mut head);
+        head = join_subtrees(&mut head_next, &mut head).expect("Failed to join subtrees");
```

Comment:
Can we specify why we don't expect this to panick? Or is that dependent on the caller of `build_root_node`?

Reply:
Should be fixed now, I did refactor the whole function across several commits.